src/guardian/duplicates/keys-duplication-checker.service.ts

```diff
@@ -167,37 +235,84 @@ export class KeysDuplicationCheckerService {
     return events;
   }
 
-  private findOriginalEvent(events: SigningKeyEvent[]): SigningKeyEvent {
-    return events.reduce(
-      (prev, curr) =>
-        prev.blockNumber < curr.blockNumber ||
-        (prev.blockNumber === curr.blockNumber && prev.logIndex < curr.logIndex)
-          ? prev
-          : curr,
-      events[0],
+  private getOperatorsWithoutEvents(
+    operators: string[],
+    events: SigningKeyEvent[],
+  ): string[] {
+    const eventOperators = new Set(
+      events.map((event) => `${event.moduleAddress}-${event.operatorIndex}`),
```

Comment:
small improvement: put `${event.moduleAddress}-${event.operatorIndex}` in a separate method